packages/swap/src/helpers/orders/makeOrderCalculations.ts

```diff
@@ -0,0 +1,126 @@
+  orderData: Swap.CreateOrderData
+  pools: Array<Swap.Pool>
+  sellAmount: Balance.Amount
+  buyAmount: Balance.Amount
+  lpTokenHeld: Balance.Amount
+  slippage: number
+  primaryTokenId: Balance.TokenInfo['id']
+  // TODO: guessing that later it will boils down to 2/3 scenarios
+  action:
+    | SwapCreateOrderActionType.SellQuantityChanged
+    | SwapCreateOrderActionType.SellTokenIdChanged
+    | SwapCreateOrderActionType.BuyQuantityChanged
+    | SwapCreateOrderActionType.BuyTokenIdChanged
+    | SwapCreateOrderActionType.SlippageChanged
+    | SwapCreateOrderActionType.PoolPairsChanged
+    | SwapCreateOrderActionType.OrderTypeChanged
+}>) => {
+  const result: Array<SwapOrderCalulation> = []
+
+  const isSellZero = Quantities.isZero(orderData.amounts.sell.quantity)
+  const isBuyZero = Quantities.isZero(orderData.amounts.buy.quantity)
+  const isLimit = orderType === 'limit'
+
+  for (const pool of pools) {
+    let buy: Balance.Amount | undefined
+    if (action === SwapCreateOrderActionType.SellQuantityChanged) {
+      buy = getBuyAmount(
+        pool,
+        orderData.amounts.sell,
+        isLimit ? orderData.limitPrice : undefined,
+      )
+    }
+    if (buy === undefined) buy = orderData.amounts.buy
+
+    let sell: Balance.Amount | undefined
+    if (action === SwapCreateOrderActionType.BuyQuantityChanged) {
+      sell = getSellAmount(
+        pool,
+        orderData.amounts.buy,
+        isLimit ? orderData.limitPrice : undefined,
+      )
+    }
+    if (sell === undefined) sell = orderData.amounts.sell
+
+    console.log(buy, sell, isBuyZero, isSellZero)
+
+    // TODO: if any side is zero set as 0
+    let price: string = ''
+    const marketPrice = getMarketPrice(pool, orderData.amounts.sell)
+    if (orderType === 'market') {
+      price = marketPrice
+    } else {
+      // NOTE: while editing should never receive undefined, otherwise it will replace with market price
+      price = orderData.limitPrice ?? marketPrice
+    }
+
+    const buyAmountWithSlippage: Balance.Amount = {
+      quantity: getQuantityWithSlippage(
+        orderData.amounts.buy.quantity,
+        orderData.slippage,
+      ),
+      tokenId: orderData.amounts.buy.tokenId,
+    }
+
+    const liquidityFee: Balance.Amount = getLiquidityProviderFee(
+      pool.fee,
+      orderData.amounts.sell,
+    )
+
+    const frontendFeeInfo = getFrontendFee({
+      sell: orderData.amounts.sell,
+      buy: orderData.amounts.buy,
+      lpTokenHeld,
+      primaryTokenId: primaryTokenId,
+      // TODO: implement after adding the pair to ADA in the state
+      // sellInPrimaryTokenValue: orderData.sellInPrimaryTokenValue,
+      // buyInPrimaryTokenValue: orderData.buyInPrimaryTokenValue,
+    })
+
+    const orderCalculation: SwapOrderCalulation = {
+      cost: {
+        batcherFee: pool.batcherFee,
+        deposit: pool.deposit,
+        frontendFeeInfo,
+        liquidityFee,
+      },
+      buyAmountWithSlippage,
+      sell: {
+        price,
```

Comment:
Where will we use this price in codebase ?

Reply:
hey @lisicky on UI, when type market this would be the pool price A/B, when limit is the user input, maybe split in 2? pricePool ("market"), priceByType (dynamic/UI)

Reply:
it will be used as input to calculate the other prices, withFees, etc...